apps/api/src/test-data/validateTestData.spec.ts

```diff
@@ -19,90 +22,177 @@ export type InMemorySnapshotOfDTOs = {
+                                 */
+                                .filter((id) => !isNullOrUndefined(id));
 
-                        const numberOfIds = allIds.length;
+                            const numberOfIds = allIds.length;
 
-                        const numberOfUniqueIds = [...new Set(allIds)].length;
+                            const numberOfUniqueIds = [...new Set(allIds)].length;
 
-                        expect(numberOfUniqueIds).toEqual(numberOfIds);
-                    });
+                            expect(numberOfUniqueIds).toEqual(numberOfIds);
+                        });
 
-                    const entityValidator = getValidatorForEntity(ResourceType);
+                        const entityValidator = getValidatorForEntity(ResourceType);
 
-                    models.forEach((dto, index) => {
-                        describe(`${ResourceType}(dto # ${index + 1})`, () => {
-                            it(`should have an id`, () => {
-                                expect(isStringWithNonzeroLength(dto.id)).toBe(true);
-                            });
-                            it(`should satisfy invariant validation`, () => {
-                                const validationResult = entityValidator(dto);
+                        models.forEach((dto, index) => {
+                            describe(`${ResourceType}(dto # ${index + 1})`, () => {
+                                it(`should have an id`, () => {
+                                    expect(isStringWithNonzeroLength(dto.id)).toBe(true);
+                                });
+                                it(`should satisfy invariant validation`, () => {
+                                    const validationResult = entityValidator(dto);
 
-                                expect(isValid(validationResult)).toBe(true);
+                                    expect(isValid(validationResult)).toBe(true);
+                                });
                             });
                         });
                     });
+
+                    const testDataInDatabaseFormat =
+                        // Use `collectionNames` not `resourceTypes` as keys
+                        Object.entries(testData).reduce(
+                            (acc, [key, models]) => ({
+                                ...acc,
+                                [getArangoCollectionIDFromResourceType(key as ResourceType)]:
+                                    models.map((model) => mapEntityDTOToDatabaseDTO(model)),
+                            }),
+                            {}
+                        );
+
+                    // TODO move this to a config- better yet avoid this whole write!
+                    const testDataFilePath = `${process.cwd()}/scripts/arangodb-docker-container-setup/docker-container-scripts/test-data/testData.json`;
+
+                    const numberOfSpacesToIndent = 4;
+
+                    writeFileSync(
```

Comment:
We will need to either
- write a second file with the edge connection test data
- update the format of the `json` output
The latter is a breaking change to @yaanahuu2 's startup script. We need to discuss this sometime.

Reply:
It would be interesting to see if we could put it all in the one JSON file, happy to update the script.

Reply:
Ok. I will make this update and add a story for you to update your script.

Reply:
Also, I like this approach. Let's make our dummy data .json file a representation of our `InMemorySnapshot`. In principle, we could dump our actual production database to JSON and then already have the tools to revive it! Even cooler, we could load the thing in memory and do an in-memory implementation of our repositories just for SAG.